roles/postgres/tasks/main.yml

```diff
@@ -125,16 +147,25 @@
       - status.phase=Running
   register: old_postgres_pod
 
-- name: Create Database if no database is specified
-  k8s:
-    apply: true
-    definition: "{{ lookup('template', 'postgres.yaml.j2') }}"
-  when: managed_database
+- block:
+    - k8s_status:
+        api_version: "{{ api_version }}"
+        kind: "{{ kind }}"
+        name: "{{ ansible_operator_meta.name }}"
+        namespace: "{{ ansible_operator_meta.namespace }}"
+        conditions:
+          - type: Database-Ready
```

Comment:
Maybe Database-Readiness would be better,
Database-Ready seems like it is ready already